apple/swift-docc
# Resolve dependency symbol reference in declarations and relationships

Bug/issue #, if applicable: rdar://116085974

## Summary

This improves the support for external documentation dependencies by resolving symbol references found in declarations and relationships in the local symbol graph files.

> [!IMPORTANT]
> There is more work needed to support this end-to-end. If you run this now you'll notice that the external links are 404s in the browser unless you host both archives on the same server.

## Dependencies

None

## Testing

To test this you need a project with two documentable targets. 

- Build documentation for one of the targets and pass `--enable-experimental-external-link-support` to enable linking to this documentation.

- Check that the output directory contains a "linkable-entities.json" file and a "link-hierarchy.json" file. (This hasn't changed since #710 but can help verify that the first target is configured correctly)

- In the other target, refer to symbols in the first target in the source code. For example:
  - Make a public subclass of a class from the first target
  - Conform a public type to a protocol from the first target
  - Make a public protocol that extends a protocol from the first target
  - Use a type from the first target as the type of a public property in the other target, `public var someProperty: First.Something`
  - Use a type from the first target as a parameter in the other target, `public func doSomething(with value: First.Something)`
  - Use a type from the first target as a return value in the other target, `public func doSomething() -⁠> First.Something`

- Build documentation for the other target and pass `--dependency /path/to/first-target.doccarchive`.

- Preview the other target's documentation and navigate to the pages that reference types from the first target.

  - Tokens that refer to types in the first target should display as links in the declarations and should be clickable.
    > Note: Unless the local server hosts both archives the link will be a 404 when clicked.
 
  - Types from the first target that are mentioned in "Inherits From", "Conforms To" and other relationship sections should display as links in the declarations and should be clickable.
    > Note: Unless the local server hosts both archives the link will be a 404 when clicked.

## Checklist

Make sure you check off the following items. If they cannot be completed, provide a reason.
